### ISSUE TITLE
Can't run multiple code actions on the same line

### ISSUE THREAD

**Issue by muirdm** [emacs-lsp/lsp-mode#1956]
**Describe the bug**

With two identically named code actions on the same line, I'm not able to execute the second code action no matter which I choose in the `lsp-execute-code-action` menu.

**To Reproduce**
Using gopls on master:
```go
package main

type foo struct{ i int }

func main() {
	_, _ = foo{}, foo{}
}
```

Move cursor to `_, _ = foo{}, foo{}` line, run `(lsp-execute-code-action)`. You should see two actions named "Fill foo with default values". Select either one and it will fill in the first `foo{}`.

**Expected behavior**
One of the code actions should fill in the second `foo{}`.

**Which Language Server did you use**
gopls

**OS**
Mac

```
[Trace - 02:06:26 PM] Received response 'textDocument/codeAction - (2562)' in 20ms.
Result: [
  {
    "command": {
      "arguments": [
        "file:///Users/muir/scratch/foo/foo.go",
        {
          "end": {
            "character": 16,
            "line": 8
          },
          "start": {
            "character": 15,
            "line": 8
          }
        }
      ],
      "command": "fill_struct",
      "title": "Fill foo with default values"
    },
    "edit": {
    },
    "kind": "refactor.rewrite",
    "title": "Fill foo with default values"
  },
  {
    "command": {
      "arguments": [
        "file:///Users/muir/scratch/foo/foo.go",
        {
          "end": {
            "character": 23,
            "line": 8
          },
          "start": {
            "character": 22,
            "line": 8
          }
        }
      ],
      "command": "fill_struct",
      "title": "Fill foo with default values"
    },
    "edit": {
    },
    "kind": "refactor.rewrite",
    "title": "Fill foo with default values"
  }
]
```

**Comment by yyoncho** [comment]
This is caused by the fact that we send all errors on the line to the server. It seems like we should send only the errors under the cursor.

**Comment by muirdm** [comment]
I'm not sure I understand - there are no errors in this case. Also the "arguments" are not meant to be read by LSP client (they are arguments back to the server when executing "fill_struct". I imagined the problem was lsp-mode was matching the actions by name, so no matter which action I pick it would end up executing the first code action with that name.

**Comment by yyoncho** [comment]
There are no warnings on this line?

**Comment by muirdm** [comment]
> There are no warnings on this line?

No, they are no diagnostics if that is what you mean. "fill_struct" is a utility code action that inserts default values for all the struct fields. It is meant to be triggered by the user when they want it, not to report any mistakes or errors to the user.

When I choose either of the "Fill foo with default values" code actions, lsp-mode sends this request to execute the code action:

```
[Trace - 09:44:44 PM] Sending request 'workspace/executeCommand - (2959)'.
Params: {
  "command": "fill_struct",
  "arguments": [
    "file:///Users/muir/scratch/foo/foo.go",
    {
      "end": {
        "character": 12,
        "line": 5
      },
      "start": {
        "character": 11,
        "line": 5
      }
    }
  ]
}
```

**Comment by yyoncho** [comment]
I am more interested in the request for code actions.

**Comment by muirdm** [comment]
```

[Trace - 10:07:10 PM] Sending request 'textDocument/codeAction - (2979)'.
Params: {
  "textDocument": {
    "uri": "file:///Users/muir/scratch/foo/foo.go"
  },
  "range": {
    "start": {
      "line": 5,
      "character": 12
    },
    "end": {
      "line": 5,
      "character": 12
    }
  },
  "context": {
    "diagnostics": []
  }
}
```

Gopls currently sends you all the code actions for the line ignoring the column. I think they are going to improve it. Regardless, it seems like lsp-mode should handle code actions with the same name.

I think the problem is here in lsp--select-action:
```
        (lsp--completing-read "Select code action: "
                              (seq-into actions 'list)
                              #'lsp:code-action-title
                              nil t)))))
```

lsp--completing-read transforms each action to its title, and then when determining which action the user chose it finds the first code action with that title (so the first code action will always be run even if the user selects the second action).

Obviously having two code actions with identical names is not normal and should be avoided, but it seems possible to come up, especially if the user selects a bigger region.

**Comment by kiennq** [comment]
A simple solution is to have transforming function to generate different name for duplicated item. With that, user can differentiate, and also the code too.

**Comment by yyoncho** [comment]
Proposed fix at #1958

**Comment by muirdm** [comment]
Looks great - thanks!
